src/main/java/io/numaproj/numaflow/function/reduce/ReduceDatumStream.java

```diff
@@ -1,16 +1,17 @@
 package io.numaproj.numaflow.function.reduce;
 
-import io.numaproj.numaflow.function.v1.Udfunction;
+import io.numaproj.numaflow.function.Datum;
+import io.numaproj.numaflow.function.HandlerDatum;
 
 /**
  * ReduceDatumStream is an interface which will be passed to
  * reduce handlers to read input messages.
  */
 public interface ReduceDatumStream {
     // EOF indicates the end of input
-    Udfunction.Datum EOF = Udfunction.Datum.newBuilder().setKey("EOF").build();
+    HandlerDatum EOF = new HandlerDatum();
```

Comment:
How do we distinguish an EOF `HandlerDatum` with others created by `new`? Before we have key=="EOF" as indicator.

Reply:
used a eof flag